packages/carbon-react/src/components/TextInput/TextInput.stories.js

```diff
@@ -0,0 +1,221 @@
+    const [type, setType] = useState('password');
+    const togglePasswordVisibility = () => {
+      setType(type === 'password' ? 'text' : 'password');
+    };
+    return (
+      <>
+        <TextInput.PasswordInput
+          type={type}
+          togglePasswordVisibility={togglePasswordVisibility}
+          ref={ref}
+          {...props}
+        />
+        <button type="button" onClick={() => setType('text')}>
+          Show password
+        </button>
+        <button type="button" onClick={() => setType('password')}>
+          Hide password
+        </button>
+      </>
+    );
+  }
+);
+
+const props = {
+  SharedInputProps: () => ({
+    className: 'some-class',
+    id: 'test2',
+    defaultValue: text(
+      'Default value (defaultValue)',
+      'This is not a default value'
+    ),
+    size: select('Field size (size)', sizes, undefined) || undefined,
+    labelText: text('Label text (labelText)', 'Text input label'),
+    placeholder: text('Placeholder text (placeholder)', 'Placeholder text'),
+    light: boolean('Light variant (light)', false),
+    disabled: boolean('Disabled (disabled)', false),
+    hideLabel: boolean('No label (hideLabel)', false),
+    invalid: boolean('Show form validation UI (invalid)', false),
+    invalidText: text(
+      'Form validation UI content (invalidText)',
+      'A valid value is required'
+    ),
+    warn: boolean('Show warning state (warn)', false),
+    warnText: text(
+      'Warning state text (warnText)',
+      'This will overwrite your current settings'
+    ),
+    helperText: text('Helper text (helperText)', 'Optional help text'),
+    inline: boolean('Inline variant (inline)', false),
+    onClick: action('onClick'),
+    onChange: action('onChange'),
+  }),
+  TextInputProps: () => ({
+    readOnly: boolean('Readonly variant (readOnly)', false),
+  }),
+  PasswordInputProps: () => ({
+    tooltipPosition: select(
+      'Tooltip position (tooltipPosition)',
+      ['top', 'right', 'bottom', 'left'],
+      'bottom'
+    ),
+    tooltipAlignment: select(
+      'Tooltip alignment (tooltipAlignment)',
+      ['start', 'center', 'end'],
+      'center'
+    ),
+    hidePasswordLabel: text(
+      '"Hide password" tooltip label for password visibility toggle (hidePasswordLabel)',
+      'Hide password'
+    ),
+    showPasswordLabel: text(
+      '"Show password" tooltip label for password visibility toggle (showPasswordLabel)',
+      'Show password'
+    ),
+  }),
+};
+
+TextInput.displayName = 'TextInput';
+
+export default {
+  title: 'Components/TextInput',
+  decorators: [withKnobs],
+
+  parameters: {
+    component: TextInput,
+    docs: {
+      page: mdx,
+    },
+    subcomponents: {
+      TextInputSkeleton,
+      'TextInput.PasswordInput': TextInput.PasswordInput,
+    },
+  },
+};
+
+export const Default = () => (
+  <TextInput
+    type={select('Form control type (type)', types, 'text')}
+    {...props.SharedInputProps()}
```

Comment:
I think the prop knobs need to be removed from the non-playground stories

Reply:
If i straight up remove them, then there's nothing being passed into those stories. How do fill the props?

Reply:
they would still take default props or any other hard coded props and they just won't be configurable via knobs. if I'm not mistaken only the playground stories (currently not in `carbon-react` yet from what I can tell) would have knobs

Reply:
Yeah, we can skip the playground stories for now and just use the simple examples with hardcoded props. We can go through the new storybook at a later time to add in more controls/knobs.